addons/mass_mailing/tests/test_mass_mailing.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.tests import common
+
+class TestMassMailing(common.TransactionCase):
+
+    def setUp(self):
+        super(TestMassMailing, self).setUp()
+        self.MassMailing = self.env['mail.mass_mailing']
+        self.MassMailingList = self.env['mail.mass_mailing.list']
+        self.MassMailingContact = self.env['mail.mass_mailing.contact']
+        self.ResPartner = self.env['res.partner']
+        self.TestMailingWizard = self.env['mail.mass_mailing.test']
+        self.MailStats = self.env['mail.mail.statistics']
+
+    def test_create_mass_mailing_to_customers(self):
+        all_customers = self.ResPartner.search([('email', '!=', False)]).ids
+        mass_mailing = self.MassMailing.create({
+            'name': 'Test Mass Mailing Customers',
+            'mailing_model': 'res.partner',
+            'reply_to_mode': 'email',
+            'mailing_domain': "[('id', 'in', %s)]" % all_customers[0:5],
+            'body_html': "<p>this is a test mass mailing to all customers<p>"})
+        mass_mailing_recipients = mass_mailing.get_remaining_recipients()
+        #checking that this mailing is to customers only, contact list ids should not set.
+        self.assertEqual(mass_mailing.contact_list_ids.ids, [], "Mass mailing created for customers not mailing lists")
+        #checking the recipients of mass mail based on given domain or not.
+        #didn't modified the mailing domain yet
+        for recipient in mass_mailing_recipients:
+            self.assertIn(recipient, all_customers[0:5], "Mailing domain not set properly")
```

Comment:
cannot we check this with assertEqual without iteration like :  
self.assertEqual(mass_mailing_recipients, all_customers[0:5], "Mailing domain not set properly")